File: src/main/java/io/github/tanguygab/tabadditions/shared/features/chat/mentions/CustomMention.java
package io.github.tanguygab.tabadditions.shared.features.chat.mentions;

import lombok.AllArgsConstructor;
import lombok.Getter;
import me.neznamy.tab.shared.placeholders.conditions.Condition;
import me.neznamy.tab.shared.platform.TabPlayer;
import net.kyori.adventure.sound.Sound;

@AllArgsConstructor
public class CustomMention {

    private final String input;
    private final String output;
    private final Condition condition;
    @Getter private final Sound sound;

    protected boolean matches(String msg) {
        return msg.matches(input);
    }

    protected String replace(String msg) {
        return msg.replaceAll(input,output);
    }

    public boolean isConditionMet(TabPlayer player) {
        return condition == null || condition.isMet(player);
    }

}

File: src/main/kotlin/io/github/tanguygab/tabadditions/shared/features/chat/mentions/CustomMention.kt
package io.github.tanguygab.tabadditions.shared.features.chat.mentions

import me.neznamy.tab.shared.placeholders.conditions.Condition
import me.neznamy.tab.shared.platform.TabPlayer
import net.kyori.adventure.sound.Sound

data class CustomMention(
    private val input: String?,
    private val output: String?,
    private val condition: Condition?,
    val sound: Sound
) {
    fun matches(msg: String) = msg.matches(input!!.toRegex())
    fun replace(msg: String) = msg.replace(input!!.toRegex(), output!!)
    fun isConditionMet(player: TabPlayer) = condition?.isMet(player) ?: true
}
